### PR TITLE
Feature/file manager

### PR DESCRIPTION
**Pull request by MaksGovor**
- [ ] all existing tests pass (npm t)
- [ ] tests that cover the new functionality are written and passed
- [ ] the style and format of the code corresponds to the style of the organization (npm run fmt)

### CODE REVIEW

File: src/service/files.service.ts [hunk 1]
```diff
@@ -19,15 +20,11 @@ export class FilesService {
...
-      const files = await this.fileRepository.getAllUserFiles(user.id);
-      return files;
-    } catch (err) {
-      return Promise.reject(err);
-    }
+    const files = await this.fileRepository.getAllUserFiles(user.id);
+    return files;
   }
 
   public async writeFileUser(
```
Comment (tedi4t) [inline src/service/files.service.ts:30]:
```suggestion
  public async upsertFile(
```

File: src/service/files.service.ts [hunk 2]
```diff
@@ -37,35 +34,45 @@ export class FilesService {
...
-        this.logger.log(err);
-        return Promise.reject(err);
-      }
-    }
+    if (!fs.existsSync(dirPath)) await fsp.mkdir(dirPath);
+
+    const currentPath = this.getFilePath(dirPath, fileName);
 
-    const currentPath = path.join(dirPath, fileName);
     const file = new File(currentPath, buffer.length, user.id);
```
Comment (tedi4t) [inline src/service/files.service.ts:41]:
```suggestion
    const file = await this.fileRepository.getByPath(file.filePath);
```

File: src/service/files.service.ts (hunk 1 above)
Comment (MaksGovor) [reply src/service/files.service.ts]: Thanks, it's fixed

File: src/service/files.service.ts (hunk 2 above)
Comment (MaksGovor) [reply src/service/files.service.ts]: Thanks, it's fixed

**Comment by MaksGovor** [comment]
Reworked the query structure from filenames to their id. @tedi4t review please

File: src/controller/files.controller.ts [hunk 3]
```diff
@@ -61,31 +72,45 @@ export class FilesController {
...
           user
         );
         return {
+          id: file.id,
           name: getFileName(file.filePath),
           size: bytesToSize(file.fileSize),
         };
       } catch (err) {
-        this.logger.error(err);
         throw new InternalServerErrorException(err);
```
Comment (tedi4t) [inline src/controller/files.controller.ts:86]: It will throw anyway

File: src/repository/file.repository.ts [hunk 4]
```diff
@@ -64,43 +62,37 @@ export class FileRepository {
...
+  public async updateFile(id: number, fileSize: number): Promise<File> {
+    const { raw } = await this.manager
+      .createQueryBuilder()
+      .update(FileEntity)
+      .set({ fileSize })
+      .where('id = :id', { id })
+      .returning('*')
+      .execute();
+
+    return (await this.getById(raw[0].id)) as File;
```
Comment (tedi4t) [inline src/repository/file.repository.ts:74]:
```suggestion
    return (await this.getById(id)) as File;
```

File: src/repository/file.repository.ts [hunk 5]
```diff
@@ -64,43 +62,37 @@ export class FileRepository {
...
-          fileSize: newFile.fileSize,
-        })
-        .where('id = :id', { id })
-        .execute();
-
-      return (await this.getById(raw[0].id)) as File;
-    }
-    return await this.insertFile(newFile);
+  public async updateFile(id: number, fileSize: number): Promise<File> {
+    const { raw } = await this.manager
```
Comment (tedi4t) [inline src/repository/file.repository.ts:66]:
```suggestion
    await this.manager
```

File: src/repository/file.repository.ts [hunk 6]
```diff
@@ -64,43 +62,37 @@ export class FileRepository {
...
+      .createQueryBuilder()
+      .update(FileEntity)
+      .set({ fileSize })
+      .where('id = :id', { id })
+      .returning('*')
```
Comment (tedi4t) [inline src/repository/file.repository.ts:71]:
```suggestion
```

File: src/repository/file.repository.ts [hunk 7]
```diff
@@ -64,43 +62,37 @@ export class FileRepository {
...
+  public async deleteFileById(id: number): Promise<File> {
+    const { raw } = await this.manager
+      .createQueryBuilder()
+      .delete()
+      .from(FileEntity)
+      .where('id = :id', { id })
+      .returning('*')
+      .execute();
+
+    return raw[0] as File;
```
Comment (tedi4t) [inline src/repository/file.repository.ts:86]: same as for update

File: src/service/files.service.ts [hunk 8]
```diff
@@ -19,59 +20,77 @@ export class FilesService {
...
+
+  private getFilePath(dirPath: string, fileName: string) {
+    const filePath = path.join(dirPath, fileName);
+    if (!filePath.startsWith(dirPath)) {
+      throw new ApplicationError('Attempt at path traversal attack');
+    }
+    return filePath;
+  }
+
+  private ensureUserFile(file: File, user: User): boolean {
```
Comment (tedi4t) [inline src/service/files.service.ts:93]:
```suggestion
  private ensureFileBelongsToUser(file: File, user: User): boolean {
```

**Comment by MaksGovor** [comment]
All resolved

**Pull request merged**